src/mqtt/client.rs

```diff
@@ -911,7 +911,53 @@ impl asynch::Connection for EspAsyncMqttConnection {
     }
 }
 
-static ERROR: EspError = EspError::from_infallible::<ESP_FAIL>();
+
+#[derive(Debug)]
+pub enum MqttClientError {
+    None,
+    TcpTransport {
+        esp_tls_last_esp_err: i32,
+        esp_tls_stack_err: i32,
+        esp_tls_cert_verify_flags: i32,
+    },
+    ConnectionRefused {
+        code: u32
+    },
+    SubscribeFailed,
+    UnknownError { code: u32 }
+}
+
+
+#[repr(transparent)]
+pub struct EspMqttEventErrorWrapper(esp_mqtt_error_codes_t);
```

Comment:
Can you please add 1-2 sentences about why this struct exists (e.g. lifetime constraints, can be translated to rust typed error with `resolve_error`).